src/System.Devices.Gpio/System.Devices.Gpio/RaspberryPiDriver.cs

```diff
@@ -300,10 +301,77 @@ public override void Dispose()
 
         public override void SetPinMode(int pin, GpioPinMode mode)
         {
-            if (pin < 0 || pin > 53) throw new ArgumentOutOfRangeException(nameof(pin));
-            if (mode < 0 || mode > GpioPinMode.Output) throw new ArgumentOutOfRangeException(nameof(mode));
+            if (pin < 0 || pin >= PinCount) throw new ArgumentOutOfRangeException(nameof(pin));
+            if (mode < 0 || mode > GpioPinMode.InputPullUp) throw new ArgumentOutOfRangeException(nameof(mode));
```

Comment:
this seems fragile.  Why don't you just add a default case to your switch below that throws for an unrecognized mode?

Reply:
Yes, I like that much more.